crates/nix_rs/src/flake/metadata.rs

```diff
@@ -0,0 +1,24 @@
+use crate::command::{NixCmd, NixCmdError};
+
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+use super::url::FlakeUrl;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FlakeMetadata {
+    pub path: PathBuf,
+}
+
+/// Runs `nix flake metadata json` in Rust
+pub async fn from_nix(cmd: &NixCmd, flake_url: &FlakeUrl) -> Result<FlakeMetadata, NixCmdError> {
+    let json = cmd
```

Comment:
```suggestion
    let metadata = cmd
```

Obviously this value is not a JSON string.